packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -581,17 +577,45 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
         }
     }
 
+    /**
+     * TODO
+     */
+    public dispose(): void {
+        if (this._closed) {
+            return;
+        }
+        this._closed = true;
+
+        this.connectionManager.dispose();
+        this.clearQueues();
+
+        this.emit("dispose");
+        this.removeAllListeners();
+    }
+
     /**
      * Closes the connection and clears inbound & outbound queues.
      */
     public close(error?: ICriticalContainerError): void {
-        if (this.closed) {
+        if (this._closed) {
```

Comment:
I'm a bit lost here. You are using same flag and doing mostly same cleanup work. And in general, this layer seems not to care about the difference. So it feels like it should either not care directly, or care only indirectly (through some argument flag that it simply passes around, but otherwise it has only one method for disposition).